utils/state_test.go

```diff
@@ -0,0 +1,142 @@
+package utils
+
+import (
+	"errors"
+	"testing"
+)
+
+// Valid json serialized state
+var validJSON = `[
+	{
+	 "name": "TEST",
+	 "cypher": "CiQArcZm2NqPRMBD38xkbUt0LDB3UPKTDq9bRPmZKRTw2B3Zv/ESOAClhb3LwQupwOSMn9K/GrBKlcBRwZorSqHFrKkM0i2yXjMYRG/hDgta2x4otKuAnpoxVCSxRiNY",
+	 "created": "2019-09-30T12:19:20.797374136-05:00",
+	 "encoding": "base64"
+	}
+   ]`
+
+// Valid Json Many
+var validManyJSON = `[
+ {
+  "name": "TEST",
+  "cypher": "CiQArcZm2NqPRMBD38xkbUt0LDB3UPKTDq9bRPmZKRTw2B3Zv/ESOAClhb3LwQupwOSMn9K/GrBKlcBRwZorSqHFrKkM0i2yXjMYRG/hDgta2x4otKuAnpoxVCSxRiNY",
+  "created": "2019-09-30T12:19:20.797374136-05:00",
+  "encoding": "base64"
+ },
+ {
+  "name": "TEST2",
+  "cypher": "CiQArcZm2JwpsNvf4q9Xbyx2nfMfhs/+TqWNnqZpFDD7o+QZ6fESNQClhb3LRJmvtCS481wCu1yWtq2M7eRHQx0Aj3xL3EHJwEexT2CZGXSZIBlQrfFI7QG07zPS",
+  "created": "2019-09-30T15:42:25.872448036-05:00",
+  "encoding": "base64"
+ }
+]`
+
+// missing the closing bracket
+var invalidJSON = `[
+	{
+	 "name": "TEST",
+	 "cypher": "CiQArcZm2NqPRMBD38xkbUt0LDB3UPKTDq9bRPmZKRTw2B3Zv/ESOAClhb3LwQupwOSMn9K/GrBKlcBRwZorSqHFrKkM0i2yXjMYRG/hDgta2x4otKuAnpoxVCSxRiNY",
+	 "created": "2019-09-30T12:19:20.797374136-05:00",
+	 "encoding": "base64"
+	}
+`
+
+// valid JSON, no data.
+var emptyJSON = `[{}]`
+
+// test table
+var validStateTests = []struct {
+	name  string
+	sm    StateManager
+	value []string
+}{
+	{"valid json", CreateTestStateManager(validJSON, nil), []string{"CiQArcZm2NqPRMBD38xkbUt0LDB3UPKTDq9bRPmZKRTw2B3Zv/ESOAClhb3LwQupwOSMn9K/GrBKlcBRwZorSqHFrKkM0i2yXjMYRG/hDgta2x4otKuAnpoxVCSxRiNY"}},
+	{"valid json many", CreateTestStateManager(validManyJSON, nil), []string{"CiQArcZm2NqPRMBD38xkbUt0LDB3UPKTDq9bRPmZKRTw2B3Zv/ESOAClhb3LwQupwOSMn9K/GrBKlcBRwZorSqHFrKkM0i2yXjMYRG/hDgta2x4otKuAnpoxVCSxRiNY", "CiQArcZm2JwpsNvf4q9Xbyx2nfMfhs/+TqWNnqZpFDD7o+QZ6fESNQClhb3LRJmvtCS481wCu1yWtq2M7eRHQx0Aj3xL3EHJwEexT2CZGXSZIBlQrfFI7QG07zPS"}},
+	{"emtpy set json", CreateTestStateManager(emptyJSON, nil), []string{""}},
+}
+
+// test-table runner
+func TestStateManagement(t *testing.T) {
+	for _, tt := range validStateTests {
+		t.Run(tt.name, func(t *testing.T) {
+			secrets, err := ReadState(tt.sm)
+
+			if err != nil {
+				t.Fatalf("Expected no errors, got %s", err)
+			}
+			for i, sec := range secrets {
+				if sec.Cyphertext != tt.value[i] {
+					t.Fatalf("Expected Cyphertext value %s but got %s", tt.value, sec.Cyphertext)
```

Comment:
I believe the `Fatalf`s here would be better off being `Errorf`. `Fatalf` fails the test immediately, so you'll only see the first case that fails and will have to rerun to see other failures. `Errorf` will fail the test, but will continue with the other cases.